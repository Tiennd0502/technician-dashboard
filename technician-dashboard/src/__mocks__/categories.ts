export const CATEGORIES = [
  {
    id: '01',
    label: 'Sofa Cleaning',
    image: 'https://i.ibb.co/dKnNFV1/category01.png',
  },
  {
    id: '02',
    label: 'Bathroom',
    image: 'https://i.ibb.co/Qdtfx8N/category02.png',
  },
  {
    id: '03',
    label: 'Best Control',
    image: 'https://i.ibb.co/dDLCwNf/category03.png',
  },
  {
    id: '04',
    label: 'View All',
    image: 'https://i.ibb.co/cDz3H5Q/category04.png',
  },
];
