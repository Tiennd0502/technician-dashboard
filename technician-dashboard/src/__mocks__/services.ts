export const SERVICE = {
  id: '01',
  name: 'Carpenter',
  price: 45,
  image: 'https://i.ibb.co/7WfPDMJ/service01.png',
};

export const SERVICES = [
  {
    id: '01',
    name: 'Carpenter',
    price: 45,
    image: 'https://i.ibb.co/7WfPDMJ/service01.png',
  },
  {
    id: '02',
    name: 'Ac Service & repair',
    price: 25,
    image: 'https://i.ibb.co/G069RKZ/service02.png',
  },
  {
    id: '03',
    name: 'Kitchecn Deep Cleaning',
    price: 30,
    image: 'https://i.ibb.co/nw3YSHS/service03.png',
  },
  {
    id: '04',
    name: 'Electrician',
    price: 50,
    image: 'https://i.ibb.co/Cvnw8GY/service04.png',
  },
];
