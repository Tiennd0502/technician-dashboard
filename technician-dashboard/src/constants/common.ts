export const LOGO_URL = '/technician.svg';

export const AVATAR_DEFAULT = '/empty-avatar.svg';

export const PRODUCT_STATISTICS_URL = 'https://i.ibb.co/FnrV3t6/total-product.webp';

export const TECHNICIAN_URL = 'https://i.ibb.co/qMWPNmx/technician.webp';

export const CURRENCY_CHARACTERS = {
  EURO: 'â‚¬',
};

export const TRUNCATE_STYLE = {
  textOverflow: 'ellipsis',
  overflow: 'hidden',
  whiteSpace: 'nowrap',
};
