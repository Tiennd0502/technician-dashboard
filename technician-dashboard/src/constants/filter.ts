import { SORT_TYPE } from '@/interfaces';

export const DEFAULT_PRODUCT_FILTER = {
  name: '',
  sortBy: 'name',
  order: SORT_TYPE.Asc,
};
