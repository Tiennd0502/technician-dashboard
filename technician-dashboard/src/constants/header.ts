export const HEADER_INFO = {
  user: {
    name: 'Belle',
    avatar: '/assets/images/empty-avatar.svg',
  },
  title: 'Product & Services',
};
