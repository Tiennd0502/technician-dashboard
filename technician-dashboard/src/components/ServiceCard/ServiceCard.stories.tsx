import type { Meta, StoryObj } from '@storybook/react';

import { SERVICE } from '@/__mocks__';
import { CURRENCY_CHARACTERS } from '@/constants';

import ServiceCard from '.';

const meta: Meta<typeof ServiceCard> = {
  title: 'Components/ServiceCard',
  component: ServiceCard,
  tags: ['autodocs'],
};

export default meta;
type Story = StoryObj<typeof ServiceCard>;

export const Default: Story = {
  args: { service: SERVICE },
};

export const HasCurrencyCharacter: Story = {
  args: { service: SERVICE, currencyCharacter: CURRENCY_CHARACTERS.EURO },
};
