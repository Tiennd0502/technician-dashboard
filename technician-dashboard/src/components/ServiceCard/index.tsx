import { memo } from 'react';
import { Link, Image, Spacer, Text, Flex } from '@chakra-ui/react';

// Constants
import { TRUNCATE_STYLE } from '@/constants';

// Types
import { Service } from '@/interfaces';

interface ServiceCardProps {
  service: Service;
  currencyCharacter?: string;
}

const ServiceCard = ({
  service: { id, name, image, price },
  currencyCharacter = '',
}: ServiceCardProps) => (
  <Link
    key={id}
    pos='relative'
    minH='165px'
    w='100%'
    borderRadius='md'
    overflow='hidden'
    _hover={{
      opacity: '0.7',
    }}
  >
    <Image
      src={image}
      alt={name}
      objectFit='cover'
      pos='absolute'
      zIndex='-1'
      w='100%'
      height='100%'
    />
    <Flex flexDirection='column' justify-content='flex-end' h='100%'>
      <Spacer />
      <Text
        color='text.tertiary'
        variant={{ base: 'textXs', md: 'textMd' }}
        px='2.5'
        {...TRUNCATE_STYLE}
      >
        {name}
      </Text>
      <Text
        color='text.tertiary'
        variant={{ base: 'textXs', md: 'textMd' }}
        px='2.5'
        mb='3'
      >{`${currencyCharacter}${price}`}</Text>
    </Flex>
  </Link>
);

export default memo(ServiceCard);
