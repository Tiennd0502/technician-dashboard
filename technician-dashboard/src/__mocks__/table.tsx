import { STATUS } from '@/interfaces';
import { StatusLabel } from '@/components';

export const PRODUCTS = [
  {
    id: '1',
    name: 'Washing machine',
    brand: 'Hitachi',
    service: 'Repair',
    status: 1,
  },
  {
    id: '2',
    name: 'Air conditioner',
    brand: 'LG',
    service: 'Installation',
    status: 0,
  },
  {
    id: '3',
    name: 'Smart TV',
    brand: 'OnePlus',
    service: 'Uninstallation',
    status: 1,
  },
  {
    id: '4',
    name: 'Water purifier',
    brand: 'Kent',
    service: 'Repair',
    status: 1,
  },
];

const customViewStatus = (value: string | number | boolean) => (
  <StatusLabel value={value as STATUS} />
);

export const PRODUCT_HEADER_COLUMNS = [
  {
    key: 'checkbox',
    label: 'Checkbox',
    width: {
      md: '30px',
      lg: '50px',
    },
    isCheckbox: true,
  },
  {
    key: 'name',
    label: 'Product name',
    width: {
      md: '100%',
      lg: '28%',
    },
    onSort: () => null,
  },
  {
    key: 'brand',
    label: 'Brand name',
    width: {
      md: '100%',
      lg: '17%',
    },
  },
  {
    key: 'service',
    label: 'Service',
    width: {
      md: '100%',
      lg: '15%',
    },
  },
  {
    key: 'status',
    label: 'Status',
    width: {
      md: '100%',
      lg: '18%',
    },
    customView: customViewStatus,
  },
  {
    key: 'actions',
    label: 'Actions',
    width: {
      md: '100%',
      lg: 'calc(32% - 35px)',
    },
    isAction: true,
  },
];
