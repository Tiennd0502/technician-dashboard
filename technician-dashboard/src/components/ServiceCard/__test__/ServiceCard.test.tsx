import { render } from '@testing-library/react';

import { SERVICE } from '@/__mocks__';
import { CURRENCY_CHARACTERS } from '@/constants';

import ServiceCard from '..';

describe('ServiceCard test cases', () => {
  const props = {
    service: SERVICE,
    currencyCharacter: CURRENCY_CHARACTERS.EURO,
  };

  test('should render correctly', () => {
    const { container } = render(<ServiceCard {...props} />);
    expect(container).toMatchSnapshot();
  });
});
