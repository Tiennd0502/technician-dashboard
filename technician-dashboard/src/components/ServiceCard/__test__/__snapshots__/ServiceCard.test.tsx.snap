// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServiceCard test cases should render correctly 1`] = `
<div>
  <a
    class="chakra-link css-1r22un5"
  >
    <img
      alt="Carpenter"
      class="chakra-image css-1fayruo"
      src="https://i.ibb.co/7WfPDMJ/service01.png"
    />
    <div
      class="css-1g4yje1"
      justify-content="flex-end"
    >
      <div
        class="css-17xejub"
      />
      <p
        class="chakra-text css-od0eia"
      >
        Carpenter
      </p>
      <p
        class="chakra-text css-18bew60"
      >
        â‚¬45
      </p>
    </div>
  </a>
</div>
`;
